zio-http/src/main/scala/zio/http/Route.scala

```diff
@@ -0,0 +1,158 @@
+package zio.http
+
+import zio._
+
+import zio.http.codec._
+
+/**
+ * Represents a single route, which has either handled its errors by converting
+ * them into responses, or which has polymorphic errors.
+ */
+sealed trait Route[-Env, +Err] { self =>
+  type PathInput
+
+  /**
+   * Applies the route to the specified request. The route must be defined for
+   * the request, or else this method will fail fatally. Note that you may only
+   * call this function when you have handled all errors produced by the route,
+   * converting them into responses.
+   */
+  def apply(request: Request)(implicit ev: Err <:< Response): ZIO[Env, Response, Response]
+
+  /**
+   * Handles the error of the route. This method can be used to convert a route
+   * that does not handle its errors into one that does handle its errors.
+   */
+  final def handleError(f: Err => Response): Route[Env, Nothing] =
+    self match {
+      case Route.Handled(routePattern, handler, l)       => Route.Handled(routePattern, handler, l)
+      case r @ Route.Unhandled(routePattern, handler, l) =>
+        Route.Handled(routePattern, (pi: r.PathInput) => handler(pi).mapError(f), l)
+    }
+
+  /**
+   * Returns a route that ignores any errors produced by this one, translating
+   * them into internal server errors that have no details. In order to provide
+   * useful feedback to users of the API, it's a good idea to capture some
+   * information on the failure and embed it into the response.
+   */
+  final def ignoreErrors: Route[Env, Nothing] =
```

Comment:
I liked my original name `withDefaultErrorResponse` better, with this one it is not clear what will happen on the client side (unless you read the comment).

Reply:
How about something like `internalServerErrors`?

Reply:
The thing I like about `ignoreErrors` is that it suggests you are doing a hack: just pretending like errors aren't happening. In a sense anytime a user calls this, they are throwing away valuable diagnostic information, so while it's convenient, it's usage should be discouraged in real apps.

Reply:
When I read your arguments @jdegoes, my first thought was: "Than just throw it away!". If it is there and easy to use, but the easy way is a misuse and discouraged, I think it is bad having it in the first place. If I discover the API through autocomplete and there is no implicit and no fancy input param to this method, that makes me think twice if that is what I want to use, I won't think. I will just see, that it produces the right types and solves my problem.

Reply:
Well, we also have `ZIO#ignore`, which serves its purpose.

Reply:
I still don't like that is is `ignore` :) For me ignoring errors would mean that if an error happen we do nothing - but this is not the case, we _do_ respond with an internal server error. Also `ZIO#ignore`'s result is something that never produces any result while this method creates a route that always produces a response.